### PR TITLE
Adding S3 Compatible Storage Support and Changing "timeout" to "frequency"

### PR DESCRIPTION
**Pull request by nehrman**
Hi, 
So I recreated a PR without the binary and with the last changes I made about timeout. 
My customer asked to change this as this was not very clear to him and he prefers to have "frequency" instead.

### CODE REVIEW

**Comment by Lucretius** [comment]
In general I probably won't want to rename properties based on a particular individuals desires because it would lead to a lot of churn for any existing users of the application - but in this particular case, I think renaming timeout to frequency (or interval or something like that) makes sense.  I'll see if I can't get this merged in today.

**Comment by Lucretius** [comment]
@nehrman Are you able to run the Go formatter on your PR?  It looks like the `frequency` property change is off-shifted in `config.go`

File: config/config.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ import (
 type Configuration struct {
 	Address  string      `json:"addr"`
 	Retain   int64       `json:"retain"`
-	Timeout  string      `json:"freq"`
+	Frequency string      `json:"frequency"`
```
Comment (Lucretius) [inline config/config.go:16]: Looks like the formatter needs to be run here

Comment (nehrman) [reply config/config.go:16]: Should be good

**Comment by Lucretius** [comment]
Thank you for the PR.  I've gone and merged it, and built a new release (v0.1.0) for you to use.

**Pull request merged**